src/main/java/com/mixpanel/android/viewcrawler/ViewCrawler.java

```diff
@@ -175,18 +175,23 @@ public void onActivityCreated(Activity activity, Bundle bundle) {
 
         @Override
         public void onActivityStarted(Activity activity) {
+            ConnectivityIndicator.start(mContext);
+            mCurrentActivity = activity;
```

Comment:
You could either create a sorted set, nullify the activity or create a weak reference. This will leak the app.